### ISSUE TITLE
Does ProxylessNas implementation really support optimizing inference latency?

### ISSUE THREAD

**Issue by tigert1998** [microsoft/nni#3113]
**Environment**:
- NNI version: v1.9 (including current master)
- NNI mode (local|remote|pai):
- Client OS:
- Server OS (for remote mode only):
- Python version:
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?:

**Log message**:
 - nnimanager.log: 
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**What issue meet, what's expected?**:

The most important feature of ProxylessNas is that it can **balance deployment latency and accuracy** with simple regularization parameters. But this feature is clearly missing in nni. I only found `loss = criterion(outputs, labels)` and `loss.backward()` where the criterion is just a cross-entropy loss and only applicable for image classification.

Would nni team consider adding this feature? If not, would you mind I writing this feature and pulling a request?

**How to reproduce it?**: 

**Additional information**:

**Comment by QuanluZhang** [comment]
@tigert1998 thanks for reporting this issue. yes, nni team has not supported this feature. it would be great if you can contribute this feature :)
